src/optimizing/Analyze_pattern.ml

```diff
@@ -85,14 +89,35 @@ let extract_strings_and_mvars ?lang any =
          * We do now semantic equivance on integers between values so
          * 1000 is now equivalent to 1_000 so we can't "regexpize" it.
          *)
-        | L (String (_, (str, _tok), _)) ->
```

Comment:
Keep the L (String _) case here, and do nothing, but add a comment saying
we used to extract the string, but we don't because of constant propagation, with the
example you put in the changelog.d